libredefender/fix_type.patch

```diff
@@ -0,0 +1,40 @@
+diff --git a/engine.rs b/engine.rs
+index 5f2fe24..e26bc58 100644
+--- a/engine.rs
++++ b/engine.rs
+@@ -78,7 +78,7 @@ pub struct Engine {
+ unsafe impl Send for Engine {}
+ unsafe impl Sync for Engine {}
+ 
+-fn map_scan_result(result: cl_error_t, virname: *const i8) -> Result<ScanResult, ClamError> {
++fn map_scan_result(result: cl_error_t, virname: *const u8) -> Result<ScanResult, ClamError> {

```

Comment:
Are these changes compatible with x86_64? I really don't think that it's the source of the problem, could you please explain the reasons for this patch further?